src/prommis/uky/uky_flowsheet.py

```diff
@@ -1126,33 +2041,577 @@ def display_results(m):
+        )
+        / (
+            units.convert(
+                metal_mass_frac["Sm2O3"]
+                * m.fs.leach_solid_feed.outlet.mass_frac_comp[0, "Sm2O3"]
+                * m.fs.leach_solid_feed.outlet.flow_mass[0],
+                to_units=units.kg / units.hr,
+            )
+        )
+        * 100
+    )
 
-def add_costing(flowsheet):
+    yt_recovery = value(
+        units.convert(
+            m.fs.sl_sep1.recovered_liquid_outlet.conc_mass_comp[0, "Y"]
+            * m.fs.sl_sep1.recovered_liquid_outlet.flow_vol[0],
+            to_units=units.kg / units.hr,
+        )
+        / (
+            units.convert(
+                metal_mass_frac["Y2O3"]
+                * m.fs.leach_solid_feed.outlet.mass_frac_comp[0, "Y2O3"]
+                * m.fs.leach_solid_feed.outlet.flow_mass[0],
+                to_units=units.kg / units.hr,
+            )
+        )
+        * 100
+    )
+
+    print(f"\nLeaching Aluminum recovery is {al_recovery} %")
+    print(f"Total aluminum recovery is {total_al_recovery} %")
+    # print(f"\nLeaching Calcium recovery is {ca_recovery} %")
+    # print(f"Total Calcium recovery is {total_ca_recovery} %")
+    print(f"\nLeaching Cerium recovery is {ce_recovery} %")
+    print(f"Total Cerium recovery is {total_ce_recovery} %")
+    print(f"\nLeaching Dysprosium recovery is {dy_recovery} %")
+    print(f"Total Dysprosium recovery is {total_dy_recovery} %")
+    print(f"\nLeaching Iron recovery is {fe_recovery} %")
+    print(f"Total Iron recovery is {total_fe_recovery} %")
+    print(f"\nLeaching Gadolinium recovery is {gd_recovery} %")
+    print(f"Total Gadolinium recovery is {total_gd_recovery} %")
+    print(f"\nLeaching Lanthanum recovery is {la_recovery} %")
+    print(f"Total Lanthanum recovery is {total_la_recovery} %")
+    print(f"\nLeaching Neodymium recovery is {nd_recovery} %")
+    print(f"Total Neodymium recovery is {total_nd_recovery} %")
+    print(f"\nLeaching Praseodymium recovery is {pr_recovery} %")
+    print(f"Total Praseodymium recovery is {total_pr_recovery} %")
+    print(f"\nLeaching Scandium recovery is {sc_recovery} %")
+    print(f"Total Scandium recovery is {total_sc_recovery} %")
+    print(f"\nLeaching Samarium recovery is {sm_recovery} %")
+    print(f"Total Samarium recovery is {total_sm_recovery} %")
+    print(f"\nLeaching Yttrium recovery is {yt_recovery} %")
+    print(f"Total Yttrium recovery is {total_yt_recovery} %")
+
+
+def add_costing(m):
     """
     Set the costing parameters for each unit model.
     """
     # TODO: Costing is preliminary until more unit model costing metrics can be verified
-    m = ConcreteModel()
-
-    # Add a flowsheet object to the model
-    m.fs = FlowsheetBlock(dynamic=True, time_units=units.s)
     m.fs.costing = QGESSCosting()
     CE_index_year = "UKy_2019"
 
     # Leaching costs
     # 4.2 is UKy Leaching - Polyethylene Tanks
     L_pe_tanks_accounts = ["4.2"]
     m.fs.L_pe_tanks = UnitModelBlock()
```

Comment:
Why are you creating a new unit model for this, and not just appending to the existing unit model? I.e.:

```
m.fs.leach.costing = UnitModelCostingBlock(...)
```

Costing is intended to be a plug-in to the existing models, not something that stands separate. The only create new unit models for balance-of-plant equipment that isn't captured in the main flowsheet, and even then you can probably do a separate costing block attached to the flowsheet (without a corresponding UnitModleBlock).

Reply:
So what was the consensus on this? Is it fine to keep `UnitModelBlock` for cases where balance-of-plant equipment aren't explicitly captured in the flowsheet? Or do we want to create generic unit models for these cases and directly add them to the flowsheet (tanks, process pumps, etc.). If the latter, that seems like it should be a part of a separate PR

Reply:
Balance of plant equipment should stay as they are, but maybe add a comment with each to indicate this.